File: CS/27_RemoveElement.cs
namespace LeetCodeSolutions {
    public static class RemoveElement {
        public static int Solution(int[] nums, int val) {
            int k = 0;

            for (int i = 0; i < nums.Length; i++) {
                if (nums[i] != val) {
                    nums[k] = nums[i];
                    k++;
                }
            }
            return k;
        }
        public static void RunTests() {
            Console.WriteLine("Running Length Of Remove Element Tests...");

            var testCases = new List<(int[], int, int)> {
                (new int[] {3, 2, 2, 3}, 3, 2),
                (new int[] {0, 1, 2, 2, 3, 0, 4, 2}, 2, 5),
            };

            foreach (var (nums, val, expected) in testCases) {
                try {
                    var result = Solution(nums, val);
                    Console.WriteLine(
                        $"Input: {string.Join(",", nums)} Val: {val} " +
                        $"Expected: {expected} Result: {result}"
                    );
                } catch (Exception ex) {
                    Console.WriteLine($"Test Failed: {ex.Message}");
                }
            }
        }
    }
}

File: JavaScript/27_RemoveElement.js
/**
 * @param {number[]} nums
 * @param {number} val
 * @return {number}
 */
var removeElement = function(nums, val) {
    let k = 0;

    for (let i = 0; i < nums.length; i++) {
        if (nums[i] !== val) {
            nums[k] = nums[i];
            k++;
        }
    }
    return k;
};

const testCases = [
    {
        nums: [3,2,2,3],
        val: 3,
        expected: 2
    },
    {
        nums: [0,1,2,2,3,0,4,2],
        val: 2,
        expected: 5
    }
]


testCases.forEach((testCase, i) => {
    const output = removeElement(testCase.nums, testCase.val);
    if (output === testCase.expected) {
        console.log(`Test Case ${i}: Passed. Expected: ${JSON.stringify(testCase.expected)}, Output: ${JSON.stringify(output)} `)
    } else {
        console.log(`Test Case ${i}: Failed. Expected: ${JSON.stringify(testCase.expected)}, Output: ${JSON.stringify(output)} `)
    }
});